File: 品味Java的21種設計模式/dp_java/proxy/example3/UserModel.java
package cn.javass.dp.proxy.example3;
/**
 * 描述用户数据的对象
 */
public class UserModel implements UserModelApi {
    /**
     * 用户编号
     */
    private String userId;
    /**
     * 用户姓名
     */
    private String name;
    /**
     * 部门编号
     */
    private String depId;
    /**
     * 性别
     */
    private String sex;

    public String getUserId() {
        return userId;
    }
    public void setUserId(String userId) {
        this.userId = userId;
    }
    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }
    public String getDepId() {
        return depId;
    }
    public void setDepId(String depId) {
        this.depId = depId;
    }
    public String getSex() {
        return sex;
    }
    public void setSex(String sex) {
        this.sex = sex;
    }

    @Override
    public String toString() {
        return "userId=" + userId + ",name=" + name + ",depId=" + depId + ",sex=" + sex + "\n";
    }
}

File: 品味Java的21種設計模式/dp_cpp/proxy/example3/UserModel.cpp
#include "UserModel.h"

namespace cn {
namespace javass {
namespace dp {
namespace proxy {
namespace example3 {

std::string UserModel::getUserId() {
    return userId;
}

void UserModel::setUserId(std::string userId) {
    this->userId = userId;
}

std::string UserModel::getName() {
    return name;
}

void UserModel::setName(std::string name) {
    this->name = name;
}

std::string UserModel::getDepId() {
    return depId;
}

void UserModel::setDepId(std::string depId) {
    this->depId = depId;
}

std::string UserModel::getSex() {
    return sex;
}

void UserModel::setSex(std::string sex) {
    this->sex = sex;
}

std::string UserModel::ToString() {
    return "userId=" + userId + ",name=" + name + ",depId=" + depId + ",sex=" + sex + "\n";
}
}
}
}
}
}
